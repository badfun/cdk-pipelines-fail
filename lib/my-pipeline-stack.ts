import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import { CodePipeline, CodePipelineSource, ShellStep } from 'aws-cdk-lib/pipelines'
import { MyPipelineAppStage } from './my-pipeline-app-stage'

export class MyPipelineStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    const pipeline = new CodePipeline(this, 'Pipeline', {
      pipelineName: 'MyPipeline',
      synth: new ShellStep('Synth', {
        input: CodePipelineSource.gitHub('OWNER/REPO', 'main'),
        commands: ['npm ci', 'npm run build', 'npx cdk synth'],
      }),
    })

    pipeline.addStage(
      new MyPipelineAppStage(this, 'test', {
        env: {
          account: this.node.tryGetContext('account'),
          region: 'eu-west-1',
        },
      })
    )

    pipeline.addStage(
      new MyPipelineAppStage(this, 'prod', {
        env: {
          account: this.node.tryGetContext('account2'),
          region: 'eu-west-1',
        },
      })
    )
  }
}
