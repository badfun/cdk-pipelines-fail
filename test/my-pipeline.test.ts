import * as cdk from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import * as MyPipeline from '../lib/my-pipeline-stack'

test('Snapshot test', () => {
  const app = new cdk.App()
  const stack = new MyPipeline.MyPipelineStack(app, 'MyPipelineStack', {
    env: {
      account: app.node.tryGetContext('account'),
      region: 'eu-west-1',
    }
  })

  expect(Template.fromStack(stack)).toMatchSnapshot()
})
