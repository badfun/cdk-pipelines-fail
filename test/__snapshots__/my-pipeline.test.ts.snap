// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "Pipeline9850B417": Object {
      "DependsOn": Array [
        "PipelineRoleDefaultPolicy7BDC1ABB",
        "PipelineRoleB27FAA37",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "Location": Object {
            "Ref": "PipelineArtifactsBucketAEA9A052",
          },
          "Type": "S3",
        },
        "Name": "MyPipeline",
        "RestartExecutionOnUpdate": true,
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "PipelineRoleB27FAA37",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "main",
                  "OAuthToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
                  "Owner": "OWNER",
                  "PollForSourceChanges": false,
                  "Repo": "REPO",
                },
                "Name": "OWNER_REPO",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "OWNER_REPO_Source",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "EnvironmentVariables": "[{\\"name\\":\\"_PROJECT_CONFIG_HASH\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"00ebacfb32b1bde8d3638577308e7b7144dfa3b0a58a83bc6ff38a3b1f26951c\\"}]",
                  "ProjectName": Object {
                    "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "OWNER_REPO_Source",
                  },
                ],
                "Name": "Synth",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "Synth_Output",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "EnvironmentVariables": "[{\\"name\\":\\"_PROJECT_CONFIG_HASH\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"bcd64e587a27153442c1241fb602f1c5c260696aea6bbf522df680826362f1c4\\"}]",
                  "ProjectName": Object {
                    "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Synth_Output",
                  },
                ],
                "Name": "SelfMutate",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "UpdatePipeline",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CHANGE_SET_REPLACE",
                  "Capabilities": "CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND",
                  "ChangeSetName": "PipelineChange",
                  "RoleArn": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":iam::",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":role/cdk-hnb659fds-cfn-exec-role-",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        "-eu-west-1",
                      ],
                    ],
                  },
                  "StackName": "test-LambdaStack",
                  "TemplatePath": "Synth_Output::assembly-MyPipelineStack-test/MyPipelineStacktestLambdaStack6ACFAD43.template.json",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Synth_Output",
                  },
                ],
                "Name": "Prepare",
                "RoleArn": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":role/cdk-hnb659fds-deploy-role-",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      "-eu-west-1",
                    ],
                  ],
                },
                "RunOrder": 1,
              },
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CHANGE_SET_EXECUTE",
                  "ChangeSetName": "PipelineChange",
                  "StackName": "test-LambdaStack",
                },
                "Name": "Deploy",
                "RoleArn": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":role/cdk-hnb659fds-deploy-role-",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      "-eu-west-1",
                    ],
                  ],
                },
                "RunOrder": 2,
              },
            ],
            "Name": "test",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CHANGE_SET_REPLACE",
                  "Capabilities": "CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND",
                  "ChangeSetName": "PipelineChange",
                  "RoleArn": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":iam::",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":role/cdk-hnb659fds-cfn-exec-role-",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        "-eu-west-1",
                      ],
                    ],
                  },
                  "StackName": "prod-LambdaStack",
                  "TemplatePath": "Synth_Output::assembly-MyPipelineStack-prod/MyPipelineStackprodLambdaStack14F0CA77.template.json",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Synth_Output",
                  },
                ],
                "Name": "Prepare",
                "RoleArn": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":role/cdk-hnb659fds-deploy-role-",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      "-eu-west-1",
                    ],
                  ],
                },
                "RunOrder": 1,
              },
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CHANGE_SET_EXECUTE",
                  "ChangeSetName": "PipelineChange",
                  "StackName": "prod-LambdaStack",
                },
                "Name": "Deploy",
                "RoleArn": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":role/cdk-hnb659fds-deploy-role-",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      "-eu-west-1",
                    ],
                  ],
                },
                "RunOrder": 2,
              },
            ],
            "Name": "prod",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "PipelineArtifactsBucketAEA9A052": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "PipelineArtifactsBucketPolicyF53CCC52": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "PipelineArtifactsBucketAEA9A052",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucketAEA9A052",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucketAEA9A052",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":role/cdk-hnb659fds-deploy-role-",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      "-eu-west-1",
                    ],
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucketAEA9A052",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucketAEA9A052",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "PipelineBuildSynthCdkBuildProject6BEFA8E6": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "Cache": Object {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "PipelineBuildSynthCdkBuildProjectRole231EEA2A",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"build\\": {
      \\"commands\\": [
        \\"npm ci\\",
        \\"npm run build\\",
        \\"npx cdk synth\\"
      ]
    }
  },
  \\"artifacts\\": {
    \\"base-directory\\": \\"cdk.out\\",
    \\"files\\": \\"**/*\\"
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "PipelineBuildSynthCdkBuildProjectRole231EEA2A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineBuildSynthCdkBuildProjectRoleDefaultPolicyFB6C941C": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:eu-west-1:111111111111:log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:eu-west-1:111111111111:log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:eu-west-1:111111111111:report-group/",
                    Object {
                      "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucketAEA9A052",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucketAEA9A052",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineBuildSynthCdkBuildProjectRoleDefaultPolicyFB6C941C",
        "Roles": Array [
          Object {
            "Ref": "PipelineBuildSynthCdkBuildProjectRole231EEA2A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineBuildSynthCodePipelineActionRole4E7A6C97": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::111111111111:root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineBuildSynthCodePipelineActionRoleDefaultPolicy92C90290": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineBuildSynthCodePipelineActionRoleDefaultPolicy92C90290",
        "Roles": Array [
          Object {
            "Ref": "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineRoleB27FAA37": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineRoleDefaultPolicy7BDC1ABB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucketAEA9A052",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucketAEA9A052",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":role/cdk-hnb659fds-deploy-role-",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    "-eu-west-1",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineRoleDefaultPolicy7BDC1ABB",
        "Roles": Array [
          Object {
            "Ref": "PipelineRoleB27FAA37",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineSourceOWNERREPOWebhookResource45DC7410": Object {
      "Properties": Object {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": Object {
          "SecretToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
        },
        "Filters": Array [
          Object {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "OWNER_REPO",
        "TargetPipeline": Object {
          "Ref": "Pipeline9850B417",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
    "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::111111111111:root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleDefaultPolicyE626265B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PipelineUpdatePipelineSelfMutationDAA41400",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleDefaultPolicyE626265B",
        "Roles": Array [
          Object {
            "Ref": "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineUpdatePipelineSelfMutationDAA41400": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "Cache": Object {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": "MyPipeline-selfupdate",
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "PipelineUpdatePipelineSelfMutationRole57E559E8",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"install\\": {
      \\"commands\\": [
        \\"npm install -g aws-cdk\\"
      ]
    },
    \\"build\\": {
      \\"commands\\": [
        \\"cdk -a . deploy MyPipelineStack --require-approval=never --verbose\\"
      ]
    }
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "PipelineUpdatePipelineSelfMutationRole57E559E8": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineUpdatePipelineSelfMutationRoleDefaultPolicyA225DA4E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:eu-west-1:111111111111:log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:eu-west-1:111111111111:log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:eu-west-1:111111111111:report-group/",
                    Object {
                      "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Condition": Object {
                "ForAnyValue:StringEquals": Object {
                  "iam:ResourceTag/aws-cdk:bootstrap-role": Array [
                    "image-publishing",
                    "file-publishing",
                    "deploy",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": "arn:*:iam::111111111111:role/*",
            },
            Object {
              "Action": "cloudformation:DescribeStacks",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "s3:ListBucket",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PipelineArtifactsBucketAEA9A052",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "PipelineArtifactsBucketAEA9A052",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineUpdatePipelineSelfMutationRoleDefaultPolicyA225DA4E",
        "Roles": Array [
          Object {
            "Ref": "PipelineUpdatePipelineSelfMutationRole57E559E8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
